### DISCUSSION TITLE
Is there any way to get current keyboard layout in wayfire?

### DISCUSSION THREAD

**Discussion by LinArcX** [WayfireWM/wayfire#1550] [Q&A]
When I was using dwm, i had this script to show the current keyboard layout on dwm bar:

```
#!/bin/sh
# Dependencies: xorg-setxkbmap

keyboard () {
    if [ "$IDENTIFIER" = "unicode" ]; then
        printf "%s" "$(xkblayout-state print %s | awk '{print toupper($0)}')"
    else
        printf "KEY %s" "$(xkblayout-state print %s)"
    fi
}
keyboard
```

But I can't use `xkblayout-state` in waybar. And I'm wondering is there any way to achieve the same functionality?

**Comment by stefonarch** [comment]
I asked the same question on matrix channel, response was:

>  unfortunately, no. There were proposals for such a protocol, but there has not been progress since ages ago .. https://gitlab.freedesktop.org/wlroots/wlr-protocols/-/merge_requests/31
